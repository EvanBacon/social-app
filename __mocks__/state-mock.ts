import {LogModel} from './../src/state/models/log'
import {LRUMap} from 'lru_map'
import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {ProfileViewModel} from '../src/state/models/profile-view'
import {MembersViewModel} from '../src/state/models/members-view'
import {ProfileUiModel, Sections} from '../src/state/models/profile-ui'
import {SessionServiceClient} from '@atproto/api'
import {UserAutocompleteViewModel} from '../src/state/models/user-autocomplete-view'
import {UserLocalPhotosModel} from '../src/state/models/user-local-photos'
import {SuggestedActorsViewModel} from '../src/state/models/suggested-actors-view'
import {UserFollowersViewModel} from '../src/state/models/user-followers-view'
import {UserFollowsViewModel} from '../src/state/models/user-follows-view'

export const mockedProfileStore = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: true,
  error: '',
  params: {
    actor: '',
  },
  did: 'test did',
  handle: 'testhandle',
  declaration: {
    cid: '',
    actorType: '',
  },
  creator: 'test did',
  displayName: '',
  description: '',
  avatar: '',
  banner: '',
  followersCount: 0,
  followsCount: 0,
  membersCount: 0,
  postsCount: 0,
  myState: {
    follow: '',
    member: '',
  },
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  isUser: true,
  isScene: false,
  setup: jest.fn().mockResolvedValue({aborted: false}),
  refresh: jest.fn(),
  toggleFollowing: jest.fn().mockResolvedValue({}),
  updateProfile: jest.fn(),
  // unknown required because of the missing private methods: _xLoading, _xIdle, _load, _replaceAll
} as unknown as ProfileViewModel

export const mockedMembersStore = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: true,
  error: '',
  params: {
    actor: 'test actor',
  },
  subject: {
    did: 'test did',
    handle: '',
    displayName: '',
    declaration: {
      cid: '',
      actorType: '',
    },
    avatar: undefined,
  },
  members: [
    {
      did: 'test did2',
      declaration: {
        cid: '',
        actorType: '',
      },
      handle: 'testhandle',
      displayName: 'test name',
      indexedAt: '',
    },
  ],
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  isMember: jest.fn(),
  setup: jest.fn().mockResolvedValue({aborted: false}),
  refresh: jest.fn(),
  loadMore: jest.fn(),
  removeMember: jest.fn(),
  // unknown required because of the missing private methods: _xLoading, _xIdle, _fetch, _replaceAll, _append
} as unknown as MembersViewModel

export const mockedMembershipsModel = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: true,
  error: '',
  params: {
    actor: '',
    limit: 1,
    before: '',
  },
  subject: {
    did: 'test did',
    handle: '',
    displayName: '',
    declaration: {cid: '', actorType: ''},
    avatar: undefined,
  },
  memberships: [
    {
      did: 'test did',
      declaration: {
        cid: '',
        actorType: 'app.bsky.system.actorUser',
      },
      handle: ',',
      displayName: '',
      createdAt: '',
      indexedAt: '',
      _reactKey: '',
    },
  ],
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  isMemberOf: jest.fn(),
  setup: jest.fn().mockResolvedValue({aborted: false}),
  refresh: jest.fn(),
  loadMore: jest.fn(),
  // unknown required because of the missing private methods: _xLoading, _xIdle, _fetch, _replaceAll, _append
} as unknown as MembershipsViewModel

export const mockedFeedModel = {
  isLoading: false,
  isRefreshing: false,
  hasNewLatest: false,
  hasLoaded: true,
  error: '',
  hasMore: true,
  params: {
    actor: '',
    limit: 1,
    before: '',
  },
  feed: [],
  rootStore: {} as RootStoreModel,
  feedType: 'home',
  hasContent: true,
  hasError: false,
  isEmpty: false,
  nonReplyFeed: [
    {
      _reactKey: 'item-1',
      post: {
        author: {
          handle: 'handle.test',
          displayName: 'test name',
          avatar: '',
        },
        cid: 'bafyreihkwjoy2vbfqld2lp3tv4ce6yfr354sqgp32qoplrudso4gyyjiwe',
        downvoteCount: 0,
        indexedAt: '2022-12-29T16:35:55.270Z',
        record: {
          $type: 'app.bsky.feed.post',
          createdAt: '2022-12-29T16:39:57.919Z',
          text: 'Sup',
        },
        replyCount: 0,
        repostCount: 0,
        upvoteCount: 0,
        uri: 'at://did:plc:wcizmlgv3rdslk64t6q4silu/app.bsky.feed.post/3jkzce5kfvn2h',
        viewer: {
          handle: 'handle.test',
          displayName: 'test name',
          avatar: '',
        },
      },
      reason: undefined,
      reply: undefined,
    },
  ],
  setHasNewLatest: jest.fn(),
  setup: jest.fn().mockResolvedValue({aborted: false}),
  refresh: jest.fn(),
  loadMore: jest.fn(),
  loadLatest: jest.fn(),
  update: jest.fn(),
  checkForLatest: jest.fn().mockRejectedValue('Error checking for latest'),
  // unknown required because of the missing private methods: _xLoading, _xIdle, _pendingWork, _initialLoad, _loadLatest, _loadMore, _update, _replaceAll, _appendAll, _prependAll, _updateAll, _getFeed, loadMoreCursor, pollCursor, _loadPromise, _updatePromise, _loadLatestPromise, _loadMorePromise
} as unknown as FeedModel

export const mockedNotificationsModel = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: true,
  error: '',
  params: {
    limit: 1,
    before: '',
  },
  hasMore: true,
  notifications: [],
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  setup: jest.fn().mockResolvedValue({aborted: false}),
  refresh: jest.fn(),
  loadMore: jest.fn(),
  update: jest.fn().mockResolvedValue(null),
  updateReadState: jest.fn(),
  // unknown required because of the missing private methods: _xLoading, _xIdle, _pendingWork, _initialLoad, _loadMore, _update, _replaceAll, _appendAll, _updateAll, loadMoreCursor, _loadPromise, _updatePromise, _loadLatestPromise, _loadMorePromise
} as unknown as NotificationsViewModel

export const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: 'test did',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn().mockResolvedValue({
    availableUserDomains: ['test'],
    links: {
      termsOfService: 'https://testTermsOfService',
      privacyPolicy: 'https://testPrivacyPolicy',
    },
  }),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown required because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

export const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: () => [
    {
      url: '/',
      title: '',
      index: 1,
      id: 1,
    },
  ],
  getForwardList: jest.fn(),
} as NavigationTabModel

export const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [mockedNavigationTabStore],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

export const mockedShellStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  darkMode: false,
  setDarkMode: jest.fn(),
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

export const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: 'test did',
  handle: 'test',
  displayName: 'test',
  description: 'test',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  memberships: mockedMembershipsModel,
  mainFeed: mockedFeedModel,
  notifications: mockedNotificationsModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

export const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

export const mockedProfilesStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn().mockResolvedValue({data: {}}),
  overwrite: jest.fn(),
} as ProfilesViewModel

export const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedLogStore = {
  entries: [],
  serialize: jest.fn(),
  hydrate: jest.fn(),
  debug: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
  // unknown required because of the missing private methods: add
} as unknown as LogModel

export const mockedRootStore = {
  api: {
    com: {},
    app: {
      bsky: {
        graph: {
          confirmation: {
            delete: jest.fn().mockResolvedValue({}),
          },
          getFollowers: jest.fn().mockResolvedValue({}),
          getMembers: jest.fn().mockResolvedValue({}),
        },
      },
    },
  } as unknown as SessionServiceClient,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfilesStore,
  linkMetas: mockedLinkMetasStore,
  log: mockedLogStore,
} as RootStoreModel

export const mockedProfileUiStore = {
  profile: mockedProfileStore,
  feed: mockedFeedModel,
  memberships: mockedMembershipsModel,
  members: mockedMembersStore,
  selectedViewIndex: 0,
  rootStore: mockedRootStore,
  params: {
    user: 'test user',
  },
  currentView: mockedFeedModel,
  isInitialLoading: false,
  isRefreshing: false,
  isUser: true,
  isScene: false,
  selectorItems: [Sections.Posts, Sections.PostsWithReplies, Sections.Scenes],
  selectedView: Sections.Posts,
  setSelectedViewIndex: jest.fn(),
  setup: jest.fn().mockResolvedValue({aborted: false}),
  update: jest.fn(),
  refresh: jest.fn(),
  loadMore: jest.fn(),
} as ProfileUiModel

export const mockedAutocompleteViewStore = {
  isLoading: false,
  isActive: true,
  prefix: '',
  follows: [
    {
      did: 'test did',
      declaration: {
        cid: '',
        actorType: 'app.bsky.system.actorUser',
      },
      handle: '',
      displayName: '',
      createdAt: '',
      indexedAt: '',
    },
  ],
  searchRes: [
    {
      did: 'test did',
      declaration: {
        cid: '',
        actorType: 'app.bsky.system.actorUser',
      },
      handle: '',
      displayName: '',
    },
  ],
  knownHandles: new Set<string>(),
  suggestions: [
    {
      handle: 'handle.test',
      displayName: 'Test Display',
    },
    {
      handle: 'handle2.test',
      displayName: 'Test Display 2',
    },
  ],
  // rootStore: {} as RootStoreModel,
  setup: jest.fn(),
  setActive: jest.fn(),
  setPrefix: jest.fn(),
  // unknown required because of the missing private methods: _searchPromise, _getFollows , _search
} as unknown as UserAutocompleteViewModel

export const mockedLocalPhotosStore = {
  photos: {
    node: {
      type: '',
      group_name: '',
      image: {
        filename: '',
        extension: '',
        uri: '',
        height: 1000,
        width: 1000,
        fileSize: null,
        playableDuration: 0,
      },
      timestamp: 1672847197,
      location: null,
    },
  },
  rootStore: {} as RootStoreModel,
  setup: jest.fn(),
  // unknown required because of the missing private methods: _getPhotos
} as unknown as UserLocalPhotosModel

export const mockedSuggestedActorsStore = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: false,
  error: '',
  suggestions: [
    {
      did: '1',
      declaration: {
        cid: '',
        actorType: 'app.bsky.system.actorUser',
      },
      handle: 'handle1.test',
      displayName: 'test name 1',
      description: 'desc',
      indexedAt: '',
      _reactKey: '1',
    },
    {
      did: '2',
      declaration: {
        cid: '',
        actorType: 'app.bsky.system.actorUser',
      },
      handle: '',
      displayName: 'handle2.test',
      description: 'desc',
      indexedAt: '',
      _reactKey: '2',
    },
  ],
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  setup: jest.fn().mockResolvedValue(null),
  refresh: jest.fn(),
  // unknown required because of the missing private methods: _xLoading, _xIdle, _fetch, _appendAll, _append
} as unknown as SuggestedActorsViewModel

export const mockedUserFollowersStore = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: false,
  error: '',
  params: {
    user: 'test user',
  },
  subject: {
    did: 'test did',
    handle: '',
    declaration: {cid: '', actorType: ''},
  },
  followers: [
    {
      did: 'test did',
      declaration: {cid: '', actorType: ''},
      handle: 'testhandle',
      displayName: 'test name',
      indexedAt: '',
      _reactKey: '1',
    },
    {
      did: 'test did2',
      declaration: {cid: '', actorType: ''},
      handle: 'testhandle2',
      displayName: 'test name 2',
      indexedAt: '',
      _reactKey: '2',
    },
  ],
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  setup: jest.fn(),
  refresh: jest.fn(),
  loadMore: jest.fn(),
  // unknown required because of the missing private methods: _xIdle, _xLoading, _fetch, _replaceAll, _append
} as unknown as UserFollowersViewModel

export const mockedUserFollowsStore = {
  isLoading: false,
  isRefreshing: false,
  hasLoaded: false,
  error: '',
  params: {
    user: 'test user',
  },
  subject: {
    did: 'test did',
    handle: '',
    declaration: {cid: '', actorType: ''},
  },
  follows: [
    {
      did: 'test did',
      declaration: {cid: '', actorType: ''},
      handle: 'testhandle',
      displayName: 'test name',
      indexedAt: '',
      _reactKey: '1',
    },
    {
      did: 'test did2',
      declaration: {cid: '', actorType: ''},
      handle: 'testhandle2',
      displayName: 'test name 2',
      indexedAt: '',
      _reactKey: '2',
    },
  ],
  rootStore: {} as RootStoreModel,
  hasContent: true,
  hasError: false,
  isEmpty: false,
  setup: jest.fn(),
  refresh: jest.fn(),
  loadMore: jest.fn(),
  // unknown required because of the missing private methods: _xIdle, _xLoading, _fetch, _replaceAll, _append
} as unknown as UserFollowsViewModel
