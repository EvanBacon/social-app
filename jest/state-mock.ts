import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {LRUMap} from 'lru_map'

const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: '',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn(),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown added because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: jest.fn(),
  getForwardList: jest.fn(),
} as NavigationTabModel

const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

const mockedShellStore = {
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: '',
  handle: '',
  displayName: '',
  description: '',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  // TODO: mock these models
  memberships: {
    placeholder: null,
  } as MembershipsViewModel,
  mainFeed: {
    placeholder: null,
  } as FeedModel,
  notifications: {
    placeholder: null,
  } as NotificationsViewModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

const mockedProfileStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn(),
  overwrite: jest.fn(),
} as ProfilesViewModel

const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedRootStore = {
  api: false,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfileStore,
  linkMetas: mockedLinkMetasStore,
}
