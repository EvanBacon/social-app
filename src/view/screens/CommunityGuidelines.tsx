import React from 'react'
import {View} from 'react-native'
import {useFocusEffect} from '@react-navigation/native'
import {ViewHeader} from '../com/util/ViewHeader'
import {useStores} from 'state/index'
import {ScrollView} from 'view/com/util/Views'
import {Text} from 'view/com/util/text/Text'
import {usePalette} from 'lib/hooks/usePalette'
import {s} from 'lib/styles'
import Html from '../../locale/en/community-guidelines'

export const CommunityGuidelinesScreen = () => {
  const pal = usePalette('default')
  const store = useStores()

  useFocusEffect(
    React.useCallback(() => {
      store.shell.setMinimalShellMode(false)
    }, [store]),
  )

  return (
    <View>
      <ViewHeader title="Community Guidelines" />
      <ScrollView style={[s.hContentRegion, pal.view]}>
        <View style={[s.p20]}>
          <Text type="title-xl" style={[pal.text, s.bold, s.pb20]}>
            Community Guidelines
          </Text>
          <Html />
        </View>
        <View style={s.footerSpacer} />
      </ScrollView>
    </View>
  )
}
