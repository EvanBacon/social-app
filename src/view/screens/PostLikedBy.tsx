import React from 'react'
import {useFocusEffect} from 'expo-router'
import {withAuthRequired} from 'view/com/auth/withAuthRequired'
import {ViewHeader} from '../com/util/ViewHeader'
import {PostLikedBy as PostLikedByComponent} from '../com/post-thread/PostLikedBy'
import {useStores} from 'state/index'
import {makeRecordUri} from 'lib/strings/url-helpers'
import {useLocalSearchParams} from 'expo-router'

export const PostLikedByScreen = withAuthRequired(() => {
  const store = useStores()
  const {name, rkey} = useLocalSearchParams<{name: string; rkey: string}>()
  const uri = makeRecordUri(name, 'app.bsky.feed.post', rkey)

  useFocusEffect(
    React.useCallback(() => {
      store.shell.setMinimalShellMode(false)
    }, [store]),
  )

  return (
    <>
      <ViewHeader title="Liked by" />
      <PostLikedByComponent uri={uri} />
    </>
  )
})
