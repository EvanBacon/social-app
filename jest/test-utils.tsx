import React from 'react'
import {render} from '@testing-library/react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import {RootSiblingParent} from 'react-native-root-siblings'
import {SafeAreaProvider} from 'react-native-safe-area-context'
import {RootStoreModel, RootStoreProvider} from '../src/state'
import {mockedRootStore} from './state-mock'

const WrappedComponent = ({children}: any) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <RootSiblingParent>
        <RootStoreProvider value={mockedRootStore as RootStoreModel}>
          <SafeAreaProvider>{children}</SafeAreaProvider>
        </RootStoreProvider>
      </RootSiblingParent>
    </GestureHandlerRootView>
  )
}

const customRender = (ui: any, options?: any) =>
  render(ui, {wrapper: WrappedComponent, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}
