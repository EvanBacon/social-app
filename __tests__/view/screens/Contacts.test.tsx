import React from 'react'
import {Contacts} from '../../../src/view/screens/Contacts'
import {render} from '../../../jest/test-utils'
import {mockedMeStore, mockedRootStore} from '../../../__mocks__/state-mock'

describe('Contacts', () => {
  const mockedProps = {
    navIdx: [0, 0] as [number, number],
    params: {},
    visible: true,
  }

  afterAll(() => {
    jest.clearAllMocks()
  })

  it('renders contacts page', () => {
    const {getByTestId} = render(<Contacts {...mockedProps} />)

    const title = getByTestId('contactsTitle')
    const textInput = getByTestId('contactsTextInput')

    expect(title).toBeTruthy()
    expect(textInput).toBeTruthy()
  })

  it('renders contacts page with handle profile follows list', () => {
    const {getByTestId} = render(<Contacts {...mockedProps} />, {
      ...mockedRootStore,
      me: {
        ...mockedMeStore,
        handle: 'test',
      },
    })
    expect(getByTestId('followList')).toBeTruthy()
  })

  it('matches snapshot', () => {
    const page = render(<Contacts {...mockedProps} />)
    expect(page).toMatchSnapshot()
  })
})
