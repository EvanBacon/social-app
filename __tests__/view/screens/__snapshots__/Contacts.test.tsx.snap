// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "letterSpacing": 0.25,
            },
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            },
          ]
        }
        testID="contactsTitle"
      >
        Contacts
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#f8f3f3",
            "borderRadius": 4,
            "flexDirection": "row",
            "marginBottom": 6,
            "marginHorizontal": 10,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <
          icon="magnifying-glass"
          size={16}
          style={
            Object {
              "color": "#645454",
              "marginRight": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Search"
          placeholderTextColor="#968d8d"
          style={
            Object {
              "color": "#000000",
              "flex": 1,
            }
          }
          testID="contactsTextInput"
          value=""
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "flexDirection": "row",
            "paddingBottom": 12,
            "paddingHorizontal": 14,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#000000",
            "bottom": 0,
            "height": 4,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 14,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 16,
                "letterSpacing": 0.25,
              },
              Array [
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "color": "#000000",
                },
              ],
            ]
          }
        >
          All
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 14,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 16,
                "letterSpacing": 0.25,
              },
              Array [
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "color": "#645454",
                },
              ],
            ]
          }
        >
          Following
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 14,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 16,
                "letterSpacing": 0.25,
              },
              Array [
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "color": "#645454",
                },
              ],
            ]
          }
        >
          Scenes
        </Text>
      </View>
    </View>
    <View
      testID="followList"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 8,
              "paddingVertical": 8,
            },
            Object {
              "backgroundColor": "#ec4899",
            },
            Object {
              "margin": 6,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 12,
                "height": 24,
                "justifyContent": "center",
                "marginRight": 8,
                "width": 24,
              },
              Object {
                "backgroundColor": "#d1106f",
              },
            ]
          }
        >
          <
            icon="exclamation"
            size={16}
            style={
              Object {
                "color": "#ffffff",
              }
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
              },
              Array [
                Object {
                  "flex": 1,
                  "paddingRight": 10,
                },
                Object {
                  "color": "#ffffff",
                },
              ],
            ]
          }
        >
          Failed to load feed: TypeError: this.rootStore.api.app.bsky.graph.getFollows is not a function
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingHorizontal": 4,
              "paddingVertical": 4,
            }
          }
          testID="tryAgainButton"
        >
          <
            icon="arrows-rotate"
            size={18}
            style={
              Object {
                "color": "#d1106f",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
